### PR TITLE
Productionize mobile aggregates job (bug 1501091)

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

File: mozaggregator/mobile.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+from moztelemetry.histogram import Histogram
+
+import warnings
+warnings.filterwarnings("always")
+
+PATH_BUCKET = environ.get('bucket', 'telemetry-parquet')
+PATH_PREFIX = 'mobile_metrics_aggregates'
+PATH_VERSION = 'v1'
+
+PROCESS_TYPES = {"parent", "content", "gpu", "dynamic", "extension"}
```
Comment (fbertsch) [inline mozaggregator/mobile.py]: This is probably not every process type. We should just get these dynamically from the pings.

File: mozaggregator/mobile.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    StructField('key', StringType(), True),
+    StructField('process', StringType(), False),
+    StructField('count', LongType(), False),
+    StructField('sum', LongType(), False),
+    StructField('histogram', MapType(StringType(), LongType(), False), False),
+])
+
+
+def get_aggregates_dataframe(spark, aggregates):
+    build_id_agg = aggregates.flatMap(lambda row: _explode(row))
```
Comment (fbertsch) [inline mozaggregator/mobile.py]: nit: is this only build_id_aggs?

File: mozaggregator/mobile.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    return state
+
+
+def _aggregate_metrics(pings):
+    trimmed = (
+        pings.map(_map_ping_to_dimensions)
+             .filter(lambda x: x))
+
+    return trimmed.aggregateByKey(
+        defaultdict(dict), _aggregate_ping, _aggregate_aggregates, 100)
```
Comment (fbertsch) [inline mozaggregator/mobile.py]: I don't think we need to set the number of partitions for this operation.

File: mozaggregator/mobile.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+    _extract_main_histograms(state, histograms, process)
+    for name, histogram in keyedHistograms.iteritems():
+        _extract_keyed_histograms(state, name, histogram, process)
+
+
+def _aggregate_ping(state, metrics):
+    if not isinstance(metrics, dict):
+        raise Exception(
+            "When is a ping not a ping? (%s)"
```
Comment (fbertsch) [inline mozaggregator/mobile.py]: We should probably change this error message :)

File: mozaggregator/mobile.py (hunk 2 above)
Comment (robhudson) [reply mozaggregator/mobile.py]: Do we also want aggregates by submission date? The current job doesn't appear to be running those.

File: mozaggregator/mobile.py (hunk 4 above)
Comment (robhudson) [reply mozaggregator/mobile.py]: Heh, I thought it was cute.

File: mozaggregator/mobile.py (hunk 3 above)
Comment (robhudson) [reply mozaggregator/mobile.py]: I can update this to work like the other aggregation job, where we are able to pass in a number from the invocation of the job. For tests we set this to a low value but for production runs it's high.

File: mozaggregator/mobile.py (hunk 1 above)
Comment (robhudson) [reply mozaggregator/mobile.py]: Would we do this once, e.g., based on the first ping we see? Or do it for every ping as we process it?

**Pull request merged**